include/zephyr/bluetooth/l2cap.h

```diff
@@ -366,6 +377,48 @@ struct bt_l2cap_chan_ops {
 	 *  @param chan The channel which was reconfigured
 	 */
 	void (*reconfigured)(struct bt_l2cap_chan *chan);
+
+#ifdef CONFIG_BT_L2CAP_SEG_RECV
+	/** @brief Handle L2CAP segments directly
+	 *
+	 *  This is an alternative to @ref bt_l2cap_chan_ops.recv. They cannot
+	 *  be used together.
```

Comment:
I assume it is on purpose that you have not guarded the `recv` with `CONFIG_BT_L2CAP_SEG_RECV=n`?